students/larry_mburu/Lesson04/mailroom.py

```diff
@@ -0,0 +1,233 @@
+#!/usr/bin/env python3
+
+import sys
+import tempfile
+import os
+from operator import itemgetter
+
+donations_per_individual = { 
+    'William Gates III' : [50000.0, 500.12], 
+    'Mark Zuckerberg' : [20000.0, 8500.99], 
+    'Jeff Bezo': [100000.0, 40000.0, 700.99], 
+    'Paul Allen': [200000.0, 1440.0, 300.00], 
+    'Bill Gates': [30000.0, 70000.0, 450.0]
+}
+
+def display_menu():
+    """
+    Display menu option to user
+    """
+    print("Welcome to Mail Automator (MA)!")
+    print()
+    print("[1] Send a Thank You!")
+    print("[2] Create a report!")
+    print("[3] Send letters to all donors")
+    print("[4] Quit!")
+    print()
+
+def letter_validator(file_tmp_dir):
+    """
+    Function just prints the content of a single letter in file, for 
+    validation only, since the send_thank_you_letter_to_all() utilizes 
+    tempfile.TempDirectory(), which does not persist files in the file system
+
+    param: file_tmp_dir: temporary directory full path name. 
+
+    return: None
+
+    """
+
+    for file in os.listdir(file_tmp_dir): 
+        print(f"[+] Listing contents of file: {file}")
+        with open(os.path.join(file_tmp_dir, file)) as fobj:
+            for line in fobj:
+                if line != '':
+                    print(line)
+        break # just one file. Avoids reading all files.
+
+def send_thank_you_letter_to_all(validate=False): 
+    """
+    Function, sends a thank you letter to all donor's in the donor DB list
+
+    param: validation: If set to True, prints the contents of single donor file, 
+    since tempfile.TempDirectory() does not persist files, in the file system. 
+    It's purely to validate that the letter / formatting was written to file
+    correctly.
+
+    return: None 
+    """
+
+    with tempfile.TemporaryDirectory() as file_tmp_dir: 
+        for donor in donations_per_individual:  
+
+            # -1, the latest donation in the donor "list"
+            email_template = f"""
+            Dear {donor}, 
+                Thank you for your kind donation of ${donations_per_individual[donor][-1]} 
+
+                it will be put to very good use. 
+
+                Sincerely,
+                    - The Cloud Squad 
+            """
+            with open(file_tmp_dir + os.sep + donor + '.txt', 'w') as fobj: 
+                fobj.write(email_template)
+
+        if validate == True: 
```

Comment:
with objects you want to use `is` instead of `==` 

you would use `is` with True/False/None and objects (later in lessons), otherwise for primitive types use `==`